administrator/components/com_installer/Model/UpdatesitesModel.php

```diff
@@ -547,7 +575,45 @@ protected function getListQuery()
 				->bind(':siteId', substr($search, 3), ParameterType::INTEGER);
 		}
 
-		// Note: The search for name, ordering and pagination are processed by the parent InstallerModel class (in extension.php).
+		if ($supported != 0)
+		{
+			switch ($supported)
+			{
+				// Show Update Sites which support Download Keys
+				case 1:
+					$supportedIDs = InstallerHelper::getDownloadKeySupportedSites($enabled);
+					break;
+
+				// Show Update Sites which are missing Download Keys
+				case -1:
+					$supportedIDs = InstallerHelper::getDownloadKeyExistsSites(false, $enabled);
+					break;
+
+				// Show Update Sites which have valid Download Keys
+				case 2:
+					$supportedIDs = InstallerHelper::getDownloadKeyExistsSites(true, $enabled);
+					break;
+			}
+
+			$supportedIDs = array_map([$db, 'quote'], $supportedIDs);
+
+			if (!empty($supportedIDs))
+			{
+				$query->where($db->qn('s.update_site_id') . ' IN (' . implode(',', $supportedIDs) . ')');
```

Comment:
Please use whereIn and quoteName.

WhereIn uses prepared statements.

```suggestion
				$query->whereIn($db->quoteName('s.update_site_id'), $supportedIDs);
```

Reply:
@HLeithner I will NOT use `quoteName`. Mandatory Joomla history lesson. `quote` and `quoteName` were introduced in Joomla! 1.6. This solved the problem of developers having to manually quote names and values in SQL statement and forgetting to do so. However, they had the drawback of being very verbose and making the code less readable. Unreadable code introduces more bugs. Hence the core developers came up with `q` and `qn` as a shorthand. These are not deprecated in Joomla! 4 so I'll continue using them because the rationale behind them has not changed in the last 10 years.

I will, however use whereIn because it's a more elegant solution, though. I keep forgetting that Joomla 4 added that feature. Thanks.

Reply:
BTW, existing code in the same model does NOT use whereIn. I will not fix that.

Reply:
it's about consistent, the complete joomla core code base (except tests) doesn't use qn or q anymore for readability. So reintroducing this make not much sense but ok your decision.

You are not forced to touch existing code and making it better code, changing unrelated code make not much sense anyway.